### PR TITLE
utility function to get latest Version w/signoff

### PR DESCRIPTION
**Pull request by Jougan-0**
**Description**

This PR fixes #

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Review by lekaf974 — APPROVED** [review]

File: utils/utils.go [hunk 1]
```diff
@@ -586,3 +588,129 @@ func Compress(src string, buf io.Writer) error {
...
+				return strings.Compare(part1, part2)
+			}
+		}
+	}
+
+	return 0
+}
+
+// Function to get all version directories sorted in descending order
+func GetAllVersionDirsSortedDesc(modelVersionsDirPath string) ([]string, error) {
```
Comment (lekaf974) [inline utils/utils.go:648]: Should we consider to add unit test on this method ?

File: utils/utils.go [hunk 2]
```diff
@@ -586,3 +588,129 @@ func Compress(src string, buf io.Writer) error {
...
+			original: versionStr,
+			dirPath:  versionDirPath,
+		})
+	}
+
+	if len(versions) == 0 {
+		return nil, fmt.Errorf("no valid version directories found in '%s'", modelVersionsDirPath)
+	}
+
+	sort.Slice(versions, func(i, j int) bool {
```
Comment (MUzairS15) [inline utils/utils.go:688]: Extract the logic and "Less" function implementation as under sorted semver version / similar.

**Review by MUzairS15 — CHANGES_REQUESTED** [review]

File: utils/utils.go (hunk 1 above)
Comment (MUzairS15) [reply utils/utils.go:648]: +1. // @Jou

File: utils/utils.go (hunk 2 above)
Comment (Jougan-0) [reply utils/utils.go:688]: Okay

**Comment by leecalcote** [comment]
We're in need of fixing the lint check errors.

**Pull request merged**